File: proyecto1/avion.cpp
#include "avion.h"

Avion::Avion(const string &marca, int capacidad, MediadorTrafico *mediator) : Aeronave(marca, capacidad, mediator)
{
}

int Avion::getAltitudMax()
{
    return this->altitudMax;
}

int Avion::getCategoria()
{
    return this->categoria;
}

int Avion::getNumMotores()
{
    return this->numMotores;
}

void Avion::printInfo()
{
    Aeronave::printInfo();
}
void Avion::obtenerDatos()
{
    cout << "Ingrese el numero de motores: ";
    cin >> this->numMotores;
    cout << endl;
    cout << "Ingrese la categoria: ";
    cin >> this->categoria;
    cout << endl;
    cout << "Ingrese la altitud maxima: ";
    cin >> this->altitudMax;
    cout << endl;
}

File: proyecto2/avion.py
from aeronave import Aeronave


class Avion(Aeronave):
    def __init__(self,id, marca, capacidad, mediator,modelo,velMax,anoFab, alitudMax, numMotores, categoría):
        super().__init__(id, marca, capacidad, mediator,modelo,velMax,anoFab)
        self.altitudMax = alitudMax
        self.numMotores = numMotores
        self.categoria = categoría

    def getAltitudMax(self):
        return self.altitudMax

    def getCategoria(self):
        return self.categoria

    def getNumMotores(self):
        return self.numMotores

    def print_info(self):
        super().printInfo()

    def obtenerDatos(self):
        try:
            self.numMotores = int(input("Ingrese el número de motores: "))
            print()

            self.categoria = int(input("Ingrese la categoría: "))
            print()

            self.altitudMax = int(input("Ingrese la altitud máxima: "))
            print()

        except ValueError:
            print("Error: Debe ingresar un número entero.")
